soundata/datasets/dcase_bioacoustic.py

```diff
@@ -0,0 +1,628 @@
+"""
+
+import os
+from typing import BinaryIO, Optional, TextIO, Tuple
+
+import librosa
+import numpy as np
+import csv
+import jams
+import glob
+import json
+
+from soundata import download_utils
+from soundata import jams_utils
+from soundata import core
+from soundata import annotations
+from soundata import io
+
+
+BIBTEX = """
+@dataset{nolasco_ines_2022_6482837,
+  author       = {Nolasco, Ines and
+                  Singh, Shubhr and
+                  Strandburg-Peshkin, Ariana and
+                  Gill, Lisa and
+                  Pamula, Hanna and
+                  Morford, Joe and
+                  Emmerson, Michael and
+                  Jensen, Frants and
+                  Whitehead, Helen and
+                  Kiskin, Ivan and
+                  Vidaña-Vila, Ester and
+                  Lostanlen, Vincent and
+                  Morfi, Veronica and
+                  Stowell, Dan},
+  title        = {{DCASE 2022 Task 5: Few-shot Bioacoustic Event 
+                   Detection Development Set}},
+  month        = mar,
+  year         = 2022,
+  publisher    = {Zenodo},
+  doi          = {10.5281/zenodo.6482837},
+  url          = {https://doi.org/10.5281/zenodo.6482837}
+}
+"""
+REMOTES = {
+    "dev": download_utils.RemoteFileMetadata(
+        filename="Development_Set.zip",
+        url="https://zenodo.org/record/6482837/files/Development_Set.zip?download=1",
+        checksum="cf4d3540c6c78ac2b3df2026c4f1f7ea",
+        # unpack_directories=["URBAN-SED_v2.0.0"],
+    ),
+    "train-classes": download_utils.RemoteFileMetadata(
+        filename="DCASE2022_task5_Training_set_classes.csv",
+        url="https://zenodo.org/record/6482837/files/DCASE2022_task5_Training_set_classes.csv?download=1",
+        checksum="abce1818ba10436971bad0b6a3464aa6",
+        # unpack_directories=["URBAN-SED_v2.0.0"],
+    ),
+    "validation-classes": download_utils.RemoteFileMetadata(
+        filename="DCASE2022_task5_Validation_set_classes.csv",
+        url="https://zenodo.org/record/6482837/files/DCASE2022_task5_Validation_set_classes.csv?download=1",
+        checksum="0c05ff0c9e1662ff8958c4c812abffdb",
+        # unpack_directories=["URBAN-SED_v2.0.0"],
+    ),
+    "eval": download_utils.RemoteFileMetadata(
+        filename="Evaluation_set_5shots.zip",
+        url="https://zenodo.org/record/6517414/files/Evaluation_set_5shots.zip?download=1",
+        checksum="5212c0e133874bba1ee25c81ced0de99",
+        # unpack_directories=["URBAN-SED_v2.0.0"],
+    ),
+}
+
+LICENSE_INFO = "Creative Commons Attribution 4.0 International"
+
+
+class Clip(core.Clip):
+    """DCASE bioacoustic Clip class
+
+    Args:
+        clip_id (str): id of the clip
+
+    Attributes:
+        audio (np.ndarray, float): path to the audio file
+        audio_path (str): path to the audio file
+        clip_id (str): clip id
+        events (soundata.annotations.Events): sound events with start time, end time, label and confidence
```

Comment:
I think that this docstring should have a `Cached properties:` subsection with `events`, and the rest of cached properties of this loader!

Reply:
done. please check